samples/java-spring-choreography-saga-quickstart/README.md

```diff
@@ -0,0 +1,182 @@
+# Choreography Saga Quickstart
+
+This quickstart sample demonstrates how to implement a Choreography Saga in Kalix.
+
+This project explores the usage of Event Sourced Entities, Value Entities and Actions.  
+To understand more about these components, see [Developing services](https://docs.kalix.io/services/) and check 
+Java-SDK [official documentation](https://docs.kalix.io/java/index.html). 
+
+## What is a Choreography Saga?
+
+A **Choreography Saga** is a distributed transaction pattern that helps you manage transactions across multiple services.
+
+In the context of event-driven applications, a **Choreography Saga** is implemented as a sequence of transactions,
+each of which publishes an event or state change notification that triggers the next operation in the saga.
+If an operation fails because it violates a business rule, then the saga can execute a series of compensating transactions
+that undo the changes made by the previous operations.
+
+In Kalix, in addition to events from Event Sourced Entities, you can also subscribe to state changes from Value Entities.
+To subscribe to events or state changes, we can use Kalix Actions with the appropriate subscription annotations.
+
+You can create a Choreography Saga to manage transactions across multiple entities in a single service, or across multiple services.
+This example implements a choreography that manages transactions across two entities in the same service.
+
+## Cross Entity Field Uniqueness
+
+A common challenge in event-sourced applications is called the _Set-Based Consistency Validation_ problem. It arises when we need to ensure that a particular field is unique across all entities in the system. For example, a user may have a unique identifier (e.g. social security number) that can be used as a unique entity ID, but may also have an email address that needs to be unique across all users in the system.
```

Comment:
It may be worth mentioning that this applies not only to ES systems but also to any system without uniqueness constraints in the DB.

Reply:
I don't think we should go that deep.